src/vs/workbench/contrib/terminal/browser/terminalActions.ts

```diff
@@ -560,8 +564,15 @@ export function registerTerminalActions() {
 				const endOfLinePreference = isWindows ? EndOfLinePreference.LF : EndOfLinePreference.CRLF;
 				text = editor.getModel().getValueInRange(selection, endOfLinePreference);
 			}
-			instance.sendText(text, true, true);
+			await instance.sendText(text, true, true);
 			await c.service.revealActiveTerminal();
+			const focusAfterRun = configurationService.getValue(TerminalSettingId.FocusAfterRun);
+			const focusTerminal = focusAfterRun === 'terminal' || (focusAfterRun === 'auto' && accessibilityService.isScreenReaderOptimized());
+			if (focusTerminal) {
+				instance.focus(true);
+			} else if (focusAfterRun === 'accessible-buffer') {
+				instance.getContribution('terminal.accessible-buffer')?.requestFocus?.();
+			}
```

Comment:
I meant stick an event on ITerminalInstance and consume it in the contribution. We never want to do something like `instance.getContribution('terminal.accessible-buffer')` as it defeats the purpose of the contribution where there's total encapsulation of the feature.

After thinking about it more, the event should be something like `onDidRunText` which can be handed for focusing the terminal in TerminalInstance and then the a11y contrib can handle focusing of the a11y buffer.

Reply:
yeah that makes sense. The `getContribution` function makes me feel like that's an appropriate thing to do though...